AnkiDroid/src/main/java/com/ichi2/anki/utils/ext/NoteType.kt

```diff
@@ -29,3 +31,19 @@ val NotetypeJson.isImageOcclusion: Boolean
     } catch (e: JSONException) {
         false
     }
+
+/**
+ * Regular expression pattern for extracting cloze text fields.
+ */
+private val clozeRegex = "\\{\\{(?:.*?:)?cloze:([^}]*)\\}\\}".toRegex()
+
+val NotetypeJson.templates: List<CardTemplateJson>
+    get() = getJSONArray("tmpls").jsonObjectIterable().map { CardTemplateJson(it) }
+
+fun NotetypeJson.getAllClozeTextFields(): List<String> {
+    val template = this.templates.singleOrNull { it.name == "Cloze" }
+        ?: throw IllegalStateException("No Cloze template found.")
+    val questionFormat = template.qfmt
+    val regex = clozeRegex
+    return regex.findAll(questionFormat).map { it.groups[1]?.value.orEmpty() }.toList()
```

Comment:
Needs testing, but what I'd expect

```suggestion
fun NotetypeJson.getAllClozeTextFields(): List<String> {
    if (!this.isCloze) {
        throw IllegalStateException("getAllClozeTextFields called on non-cloze template")
    }
    val questionFormat = templates.single().qfmt
    return clozeRegex.findAll(questionFormat).map { it.groups[1]!!.value }.toList()
}
```